sycl/include/sycl/detail/common.hpp

```diff
@@ -119,6 +119,72 @@ struct code_location {
 #define _CODELOCFW(a)
 #endif
 
+/// @brief The TLS variable into which Code Locations will be stashed
+extern thread_local detail::code_location GCodeLocTLS;
```

Comment:
This needs overall description of what is being done. Why would we need a global variable for this? I expect I'll realize after reading the rest of this PR, but we need a short summary here for the readers after this is merged.

Reply:
Done! - added more to the comments.

Reply:
Should have more documentation and the PR summary as been updated as well.